src/com/facebook/buck/android/AndroidLibraryDescription.java

```diff
@@ -47,17 +52,20 @@
 import java.nio.file.Path;
 
 public class AndroidLibraryDescription
-    implements Description<AndroidLibraryDescription.Arg>, Flavored {
+    implements Description<AndroidLibraryDescription.Arg>, Flavored,
+       ImplicitDepsInferringDescription<AndroidLibraryDescription.Arg> {
 
   public static final BuildRuleType TYPE = BuildRuleType.of("android_library");
 
   private static final Flavor DUMMY_R_DOT_JAVA_FLAVOR =
       AndroidLibraryGraphEnhancer.DUMMY_R_DOT_JAVA_FLAVOR;
 
   private final JavacOptions defaultOptions;
+  private final ScalaBuckConfig scalaBuckConfig;
 
-  public AndroidLibraryDescription(JavacOptions defaultOptions) {
+  public AndroidLibraryDescription(JavacOptions defaultOptions, ScalaBuckConfig scalaConfig) {
```

Comment:
1) `JvmLanguagesConfig` would need to include almost everything that is in language specific configs, I mean there doesn't seem to be a common subset of settings that would be enough to instantiate any `AndroidLibraryCompiler`
2) `getCompiler` uses `language` param, so it can not be created before `AndroidLibraryDescription` 

3)
Other option would be to have class providing suppliers for all configs, something like:
```
abstract class AndroidLibraryConfig {
    Supplier<ScalaBuckConfig> getScalaConfigSuplier();
    Supplier<KotlinBuckConfig> getKotlinConfigSupplier();
    ...
}
```
All methods would have default implementations throwing exceptions, so there is no need to specify all configs every time.
What do you think about that?